clients/src/main/java/org/apache/kafka/clients/admin/MemberAssignment.java

```diff
@@ -19,21 +19,25 @@
 import org.apache.kafka.common.TopicPartition;
 import org.apache.kafka.common.utils.Utils;
 
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
 import java.util.List;
 
 /**
  * A description of the assignments of a specific group member.
  */
 public class MemberAssignment {
-    private final List<TopicPartition> topicPartitions;
+    private final Collection<TopicPartition> topicPartitions;
```

Comment:
Curious why you made this change. Lists are a bit more convenient for users.

Reply:
To me, returning a List implies that ordering is important, which it's not in this case.  Using Collection gives us the freedom to change it in the future.  Most things are just as convenient on either one, right?  You can iterate using the standard for loop, iterators, etc.

Reply:
One issue with `Collection` as that `equals` is not defined. So, one should be careful about returning it in public APIs. It's OK to receive it as a parameter though.